routes/resources.js

```diff
@@ -10,27 +10,44 @@ var Schemas = require('../generators/schemas');
 
 module.exports = function (app, model, Implementation, integrator, opts) {
   var modelName = Implementation.getModelName(model);
-  var schema = Schemas.schemas[Implementation.getModelName(model)];
+  var schema = Schemas.schemas[modelName];
+
+  function addSmartFieldToRecord(record, smartFieldObject, modelName) {
+    var smartFieldValue;
+    if(smartFieldObject.get || smartFieldObject.value) {
+      if (smartFieldObject.value) {
+        logger.warn('DEPRECATION WARNING: In your ' + modelName +
+          ' Model, Smart Field value method is deprecated. ' +
+          'Please use get method instead. ');
+        smartFieldValue = smartFieldObject.value(record);
+      } else {
+        smartFieldValue = smartFieldObject.get(record);
+      }
+      if (smartFieldValue && _.isFunction(smartFieldValue.then)) {
+        return smartFieldValue.then(function (value) {
+          record[smartFieldObject.field] = value;
+        });
+      } else {
+        record[smartFieldObject.field] = smartFieldValue;
+      }
+    } else if (_.isArray(smartFieldObject.type)) {
+      record[smartFieldObject.field] = [];
+    }
```

Comment:
The `elsif` is not part of the Smart Field getter logic. I am not sur you should use it for the associations.